src/Nocarrier/Hal.php

```diff
@@ -234,10 +234,15 @@ public function setData(Array $data = null)
     /**
      * Return an array of data (key => value pairs) representing this resource.
      *
+     * @param null|string data key
      * @return array
      */
-    public function getData()
+    public function getData($key = null)
     {
+        if ($key) {
+            return isset($this->data[$key]) ? $this->data[$key] : null;
```

Comment:
If you mean creating a new method to retrieve a named resource like here, then yes. It that cause a new contract is defined for for a new method.

Another option may be to return an empty array instead of `null`. Because that is consistent with what the return value for this method specifies.